shelley/chain-and-ledger/formal-spec/update.tex

```diff
@@ -0,0 +1,396 @@
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Application Version Update transitions}
+  \begin{equation*}
+    \_ \vdash
+    \var{\_} \trans{avup}{\_} \var{\_}
+    \subseteq \powerset (\AVUpdateEnv \times \AVUpdateState \times \AVUpdate \times \AVUpdateState)
+  \end{equation*}
+  %
+  \caption{Application Version Update transition-system types}
+  \label{fig:ts-types:av-update}
+\end{figure}
+
+\begin{figure}[htb]
+  \begin{equation}\label{eq:av-update-Empty}
+    \inference[AV-Update-Empty]
+    {
+      \var{aup} = \emptyset
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{dms}\\
+      \end{array}
+      \vdash
+      \left(
+      \begin{array}{l}
+        \var{aup_s}\\
+        \var{favs}\\
+        \var{avs}\\
+      \end{array}
+      \right)
+      \trans{avup}{aup}
+      \left(
+      \begin{array}{l}
+        \var{aup_s}\\
+        \var{favs}\\
+        \var{avs}\\
+      \end{array}
+      \right)
+    }
+  \end{equation}
+
+  \nextdef
+
+  \begin{equation}\label{eq:update-no-consensus}
+    \inference[AV-Update-No-Consensus]
+    {
+      \var{aup}\neq\emptyset
+      &
+      \dom{\var{aup}}\subseteq\dom{\var{dms}}
+      \\
+      \var{aup'}\leteq\var{aup_s}\unionoverrideRight\var{aup}
+      &
+      \var{fav}\leteq\fun{votedValue_{Applications}}~\var{aup'}
+      \\
+      \var{fav}=\Nothing
+      \\
+      \var{cur}\leteq\{\var{s}\mapsto\var{v}\in\var{favs} ~\mid~ s\geq\var{slot}\}
+      \\
+      \var{avs'}\leteq\fun{newAVs}~\var{avs}~\var{cur}
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{dms}\\
+      \end{array}
+      \vdash
+      \left(
+      \begin{array}{l}
+        \var{aup_s}\\
+        \var{favs}\\
+        \var{avs}\\
+      \end{array}
+      \right)
+      \trans{avup}{aup}
+      \left(
+      \begin{array}{l}
+        \varUpdate{\var{aup'}}\\
+        \varUpdate{\var{favs}\setminus\var{cur}}\\
+        \varUpdate{\var{avs'}}\\
+      \end{array}
+      \right)
+    }
+  \end{equation}
+
+  \nextdef
+
+  \begin{equation}\label{eq:update-consensus}
+    \inference[AV-Update-Consensus]
+    {
+      \var{aup}\neq\emptyset
+      &
+      \dom{\var{aup}}\subseteq\dom{\var{dms}}
+      \\
+      \var{aup'}\leteq\var{aup_s}\unionoverrideRight\var{aup}
+      &
+      \var{fav}\leteq\fun{votedValue_{Applications}}~\var{aup'}
+      \\
+      \var{fav}\neq\Nothing
+      \\
+      \var{cur}\leteq\{\var{s}\mapsto\var{v}\in\var{favs} ~\mid~ s\geq\var{slot}\}
+      \\
+      \var{avs'}\leteq\fun{newAVs}~\var{avs}~\var{cur}
+      \\
+      s\leteq\var{slot}+\SlotsPrior
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{dms}\\
+      \end{array}
+      \vdash
+      \left(
+      \begin{array}{l}
+        \var{aup_s}\\
+        \var{favs}\\
+        \var{avs}\\
+      \end{array}
+      \right)
+      \trans{avup}{aup}
+      \left(
+      \begin{array}{l}
+        \varUpdate{\emptyset}\\
```

Comment:
Here it is kind of weird that we throw away the other proposals when there is consensus on a given application ...